### PR TITLE
Change from options to builder the metrics creation.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/bogdandrutu/openconsensus/issues/189

### CODE REVIEW

File: api/src/main/java/openconsensus/metrics/Metric.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * limitations under the License.
+ */
+
+package openconsensus.metrics;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public interface Metric {
```
Comment (songy23) [inline api/src/main/java/openconsensus/metrics/Metric.java]: Nit: javadoc

Comment (bogdandrutu) [reply api/src/main/java/openconsensus/metrics/Metric.java]: Done.

**Comment by songy23** [comment]
Please fix format.

**Comment by bogdandrutu** [comment]
@songy23 done

**Pull request merged**